dts/bindings/ethernet/ethernet,fixed-link.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2020, DENX Software Engineering GmbH
+#		      Lukasz Majewski <lukma@denx.de>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Ethernet fixed link configuration for PHY communication
+
+compatible: "ethernet,fixed-link"
```

Comment:
If you look at the standard device tree bindings for "fixed-link", there is no compatible property.. Whenever the Ethernet device is enumerated, the driver is supposed to search for the "fixed-link" node and assume that the driver uses fixed link if found.. See https://elixir.bootlin.com/linux/v4.5/source/drivers/of/of_mdio.c#L376 for details.. 

I'm not sure what exactly is the policy about introducing "zephyr specific" bindings, but this seems avoidable..

Reply:
On the beginning I wanted to follow the same pattern - i.e. to have the 'fixed-link' parsed as a subnode.
However, I've encountered the following issue:
- In Zephyr I use DT_CHILD(DT_ALIAS(eth), fixed_link) to get the information if 'fixed-link' property is available.
- However, there is the issue - without proper ethernet,fixed-link.yaml and compatible node
the parsed DTS (in zephyr/include/generated/devicetree_unfixed.h) doesn't have required
defines (for example #define DT_N_S_soc_S_ethernet_400c0004_S_fixed_link*).
Without them the DT_CHILD(DT_ALIAS(eth), fixed_link) macro resolve to 0.

Is there any way to avoid having "compatible" property and have required defines for DT_CHILD* macros?